### PR TITLE
LPS-155120 Adapt Object to support metadata fields as system fields

### PR DESCRIPTION
**Pull request by danvb112**
Related Issues: 

[LPS-155120 - [FE] Add filter to system fields in Modal to add field in layout screen](https://issues.liferay.com/browse/LPS-155120)
[LPS-155121 - [FE] Make only the label available for editing system fields in SidePanel in EditObjectField.tsx](https://issues.liferay.com/browse/LPS-155121)
[LPS-155122 - [FE] Avoid duplicating METADATA fields in Object View](https://issues.liferay.com/browse/LPS-155122)
[LPS-155123 - [FE] Avoid duplicating METADATA fields in Object Validation with Expression Builder](https://issues.liferay.com/browse/LPS-155123)

### CODE REVIEW

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by naraandrade** [comment]
Just started reviewing :)

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/Layout/context.tsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ import {BoxesVisitor} from '../../utils/visitor';
 import {TObjectField, TObjectLayout, TObjectRelationship} from './types';
 
 type TState = {
+	ffMetadataFieldsAsSystemFields: boolean;
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/Layout/context.tsx:25]: @danvb112 You don't need to receive the value for the Feature Flag from the back-end and pass it through the components. You could just use `Liferay.FeatureFlags['LPS-154872']` :)

**Review by naraandrade — CHANGES_REQUESTED** [review]

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:forward

**Comment by danvb112** [comment]
ci:forward

**Pull request closed without merging**